File: frontend/src/utils/enums/finalidadeEnum.js
const FinalidadesEnum =  Object.freeze(
		[
			{value: 'ATUALIZACAO', text:'Atualização'},
			{value: 'CADASTRO', text: 'Cadastro'},
			{value: 'DISPENSA', text: 'Dispensa'},
			{value: 'RENOVACAO', text: 'Renovação'},
			{value: 'SOLICITACAO', text: 'Solicitação'}
		]
);

const getMap = function() {

	let finalidades = new Map();

	FinalidadesEnum.forEach(finalidade => {
		finalidades.set(finalidade.value, finalidade);
	});

	return finalidades;
};

export default {FinalidadesEnum, getMap};

File: backend/src/main/java/com/configuradorlicenciamento/licenca/models/FinalidadeEnum.java
package com.configuradorlicenciamento.licenca.models;

import java.util.HashMap;
import java.util.Map;

public enum FinalidadeEnum {

    ATUALIZACAO("ATUALIZACAO", "Atualização"),
    CADASTRO("CADASTRO", "Cadastro"),
    DISPENSA("DISPENSA", "Dispensa"),
    RENOVACAO("RENOVACAO", "Renovação"),
    SOLICITACAO("SOLICITACAO", "Solicitação");

    private static final Map<String, String> LABEL = new HashMap<>();

    public final String key;
    public final String label;

    FinalidadeEnum(String key, String label) {
        this.key = key;
        this.label = label;
    }

    static {
        for (FinalidadeEnum finalidade: values()) {
            LABEL.put(finalidade.key, finalidade.label);
        }
    }

    public static String valueOfLabel(String key) {
        return LABEL.get(key);
    }

}
